runtime/queries/metricsview_comparison_toplist.go

```diff
@@ -425,13 +459,25 @@ func (q *MetricsViewComparison) buildMetricsComparisonTopListSQL(mv *runtimev1.M
 		var columnsTuple string
 		var labelTuple string
 		if dialect != drivers.DialectDruid {
-			columnsTuple = fmt.Sprintf(
-				"base.%[1]s, comparison.%[1]s AS %[2]s, base.%[1]s - comparison.%[1]s AS %[3]s, (base.%[1]s - comparison.%[1]s)/comparison.%[1]s::DOUBLE AS %[4]s",
-				safeName(m.Name),
-				safeName(m.Name+"__previous"),
-				safeName(m.Name+"__delta_abs"),
-				safeName(m.Name+"__delta_rel"),
-			)
+			if q.Having != nil {
+				// having clause needs selected columns to either be in group by or be aggregations.
+				// so adding additional sum() around measure and comparison columns
+				columnsTuple = fmt.Sprintf(
+					"sum(base.%[1]s) as %[1]s, sum(comparison.%[1]s) AS %[2]s, sum(base.%[1]s - comparison.%[1]s) AS %[3]s, sum((base.%[1]s - comparison.%[1]s)/comparison.%[1]s::DOUBLE) AS %[4]s",
+					safeName(m.Name),
+					safeName(m.Name+"__previous"),
+					safeName(m.Name+"__delta_abs"),
+					safeName(m.Name+"__delta_rel"),
+				)
+			} else {
+				columnsTuple = fmt.Sprintf(
+					"base.%[1]s, comparison.%[1]s AS %[2]s, base.%[1]s - comparison.%[1]s AS %[3]s, (base.%[1]s - comparison.%[1]s)/comparison.%[1]s::DOUBLE AS %[4]s",
+					safeName(m.Name),
+					safeName(m.Name+"__previous"),
+					safeName(m.Name+"__delta_abs"),
+					safeName(m.Name+"__delta_rel"),
+				)
+			}
```

Comment:
Is this needed because the actual aggregation happens in a sub-query? If yes, then instead of adding a new layer of aggregations, couldn't we just use a `WHERE` clause in the outer select instead? (Since a `HAVING` clause is basically equivalent to a `WHERE` clause in a wrapped query, or am I missing something?)

If the aggregation is needed, consider using `ANY_VALUE` or `LAST`, which I believe we use for such cases in other places (less ambiguous than `SUM`).

Reply:
Good point. I have updated to use an outer where clause.